In [5]:
import pandas as pd
import numpy as np

# Exploratory Data Analysis - 1st Step

    #First section for data mining.

In the below section, it's the description of the raw data as extract of the dataset for the project and the first readings for cleaning 

In [27]:
df = pd.read_csv("/home/alexg/Bedu/Project/Dataframe/Original/ks-projects-201801.csv")
df.head(5)

ID                                               name  \
0  1000002330                    The Songs of Adelaide & Abullah   
1  1000003930      Greeting From Earth: ZGAC Arts Capsule For ET   
2  1000004038                                     Where is Hank?   
3  1000007540  ToshiCapital Rekordz Needs Help to Complete Album   
4  1000011046  Community Film Project: The Art of Neighborhoo...   

         category main_category currency    deadline     goal  \
0          Poetry    Publishing      GBP  2015-10-09   1000.0   
1  Narrative Film  Film & Video      USD  2017-11-01  30000.0   
2  Narrative Film  Film & Video      USD  2013-02-26  45000.0   
3           Music         Music      USD  2012-04-16   5000.0   
4    Film & Video  Film & Video      USD  2015-08-29  19500.0   

              launched  pledged     state  backers country  usd pledged  \
0  2015-08-11 12:12:28      0.0    failed        0      GB          0.0   
1  2017-09-02 04:43:57   2421.0    failed       15      US        100.0   
2  2013-01-12 00:20:50    220.0    failed        3      US        220.0   
3  2012-03-17 03:24:11      1.0    failed        1      US          1.0   
4  2015-07-04 08:35:03   1283.0  canceled       14      US       1283.0   

   usd_pledged_real  usd_goal_real  
0               0.0        1533.95  
1            2421.0       30000.00  
2             220.0       45000.00  
3               1.0        5000.00  
4            1283.0       19500.00

    #Scanning the type of data
    Here, there is a first scanning of the data as the type of data:

In [26]:
df.dtypes

ID                    int64
name                 object
category             object
main_category        object
currency             object
deadline             object
goal                float64
launched             object
pledged             float64
state                object
backers               int64
country              object
usd pledged         float64
usd_pledged_real    float64
usd_goal_real       float64
dtype: object

In [42]:
df.shape

(378661, 15)

In [43]:
duplicate_rows_df = df[df.duplicated()]
print("number of duplicate rows: ", duplicate_rows_df.shape)

number of duplicate rows:  (0, 15)


In [57]:
df = pd.read_csv("/home/alexg/Bedu/Project/Dataframe/Original/ks-projects-201801.csv")
df.count()

ID                  378661
name                378657
category            378661
main_category       378661
currency            378661
deadline            378661
goal                378661
launched            378661
pledged             378661
state               378661
backers             378661
country             378661
usd pledged         374864
usd_pledged_real    378661
usd_goal_real       378661
dtype: int64

In [58]:
df = pd.read_csv("/home/alexg/Bedu/Project/Dataframe/Original/ks-projects-201801.csv")
print(df.isnull().sum())

ID                     0
name                   4
category               0
main_category          0
currency               0
deadline               0
goal                   0
launched               0
pledged                0
state                  0
backers                0
country                0
usd pledged         3797
usd_pledged_real       0
usd_goal_real          0
dtype: int64


    Here, we can notice the column "usd pledged" (in original form), has null data and the "name" column as well is having null data

In [59]:
df = df.dropna()
df.count()

ID                  374860
name                374860
category            374860
main_category       374860
currency            374860
deadline            374860
goal                374860
launched            374860
pledged             374860
state               374860
backers             374860
country             374860
usd pledged         374860
usd_pledged_real    374860
usd_goal_real       374860
dtype: int64

    Deleting the null data, the remaining data is 374860 out of 378661 meaning (98% of the data)

    Once deleted the null data, the next exploratory analysis will allow to undertand certain outlier data, like unique values or

In [60]:
include =['object', 'float']
df = df.dropna()
desc=df.describe(include = include)
desc

name        category main_category currency  \
count                374860          374860        374860   374860   
unique               372068             159            15       14   
top     Canceled (Canceled)  Product Design  Film & Video      USD   
freq                     13           22310         62696   292624   
mean                    NaN             NaN           NaN      NaN   
std                     NaN             NaN           NaN      NaN   
min                     NaN             NaN           NaN      NaN   
25%                     NaN             NaN           NaN      NaN   
50%                     NaN             NaN           NaN      NaN   
75%                     NaN             NaN           NaN      NaN   
max                     NaN             NaN           NaN      NaN   

          deadline          goal             launched       pledged   state  \
count       374860  3.748600e+05               374860  3.748600e+05  374860   
unique        3164           NaN               374298           NaN       5   
top     2014-08-08           NaN  1970-01-01 01:00:00           NaN  failed   
freq           702           NaN                    7           NaN  197611   
mean           NaN  4.952299e+04                  NaN  9.750538e+03     NaN   
std            NaN  1.189362e+06                  NaN  9.601094e+04     NaN   
min            NaN  1.000000e-02                  NaN  0.000000e+00     NaN   
25%            NaN  2.000000e+03                  NaN  3.100000e+01     NaN   
50%            NaN  5.500000e+03                  NaN  6.200000e+02     NaN   
75%            NaN  1.650000e+04                  NaN  4.080000e+03     NaN   
max            NaN  1.000000e+08                  NaN  2.033899e+07     NaN   

       country   usd pledged  usd_pledged_real  usd_goal_real  
count   374860  3.748600e+05      3.748600e+05   3.748600e+05  
unique      22           NaN               NaN            NaN  
top         US           NaN               NaN            NaN  
freq    292624           NaN               NaN            NaN  
mean       NaN  7.036802e+03      9.120899e+03   4.586303e+04  
std        NaN  7.864016e+04      9.131969e+04   1.158767e+06  
min        NaN  0.000000e+00      0.000000e+00   1.000000e-02  
25%        NaN  1.698000e+01      3.100000e+01   2.000000e+03  
50%        NaN  3.947650e+02      6.244950e+02   5.500000e+03  
75%        NaN  3.034425e+03      4.050895e+03   1.600000e+04  
max        NaN  2.033899e+07      2.033899e+07   1.663614e+08

In [52]:
import pandas as pd
import re
df = pd.read_csv("/home/alexg/Bedu/Project/Dataframe/Original/ks-projects-201801.csv")
df.dropna(inplace = True)
desc = df["main_category"].describe()
desc

count           374860
unique              15
top       Film & Video
freq             62696
Name: main_category, dtype: object

    #Returning the data to csv for other changes

In [65]:
df.to_csv('/home/alexg/Bedu/Project/Dataframe/ks_df_modified_python1.csv', index = False)